specs/interop/eth-lockbox.md

```diff
@@ -0,0 +1,255 @@
+
+```solidity
+function unlockETH(uint256 _value) external;
+```
+
+#### `authorizePortal`
+
+Authorizes an `OptimismPortal` to interact with the `ETHLockbox`.
+
+- Only the `ProxyAdmin` owner can call the function.
+- The function MUST emit the `PortalAuthorized` event with the `portal`.
+- The function MUST NOT allow the same `OptimismPortal` to be authorized more than once.
+
+```solidity
+function authorizePortal(address _portal) external;
+```
+
+#### `authorizeLockbox`
+
+Authorizes another `ETHLockbox` to migrate its ETH liquidity to the current `ETHLockbox`.
+
+- Only the `ProxyAdmin` owner can call the function.
+- The function MUST emit the `LockboxAuthorized` event with the `lockbox` that is being authorized.
+- The function MUST NOT allow the same `ETHLockbox` to be authorized more than once.
+
+```solidity
+function authorizeLockbox(address _lockbox) external;
+```
+
+#### `migrateLiquidity`
+
+Migrates the ETH liquidity from the current `ETHLockbox` to another `ETHLockbox`.
+
+- Only the `ProxyAdmin` owner can call the function.
+- The `ProxyAdmin` owner of the source lockbox must be the same as the `ProxyAdmin` owner of the destination lockbox.
+- The function MUST call `receiveLiquidity` from the destination `ETHLockbox`.
+- The function MUST emit the `LiquidityMigrated` event with the `lockbox` that is being migrated to.
+
+```solidity
+function migrateLiquidity(address _lockbox) external;
+```
+
+#### `receiveLiquidity`
+
+Receives the ETH liquidity from another `ETHLockbox`.
+
+- Only an authorized `ETHLockbox` can call the function.
+- The function MUST emit the `LiquidityReceived` event with the `lockbox` that is being received from.
+
+```solidity
+function receiveLiquidity() external payable;
+```
+
+#### `adminOwner`
+
+Returns the `ProxyAdmin` owner that manages the `ETHLockbox`.
+
+```solidity
+function adminOwner() external view returns (address);
+```
+
+### Events
+
+#### `ETHLocked`
+
+MUST be triggered when `lockETH` is called
+
+```solidity
+event ETHLocked(address indexed portal, uint256 amount);
+```
+
+#### `ETHUnlocked`
+
+MUST be triggered when `unlockETH` is called
+
+```solidity
+event ETHUnlocked(address indexed portal, uint256 amount);
+```
+
+#### `PortalAuthorized`
+
+MUST be triggered when `authorizePortal` is called
+
+```solidity
+event PortalAuthorized(address indexed portal);
+```
+
+#### `LockboxAuthorized`
+
+MUST be triggered when `authorizeLockbox` is called
+
+```solidity
+event LockboxAuthorized(address indexed lockbox);
+```
+
+#### `LiquidityMigrated`
+
+MUST be triggered when `migrateLiquidity` is called
+
+```solidity
+event LiquidityMigrated(address indexed lockbox);
+```
+
+#### `LiquidityReceived`
+
+MUST be triggered when `receiveLiquidity` is called
+
+```solidity
+event LiquidityReceived(address indexed lockbox);
+```
+
+## Invariants
+
+### System level invariants
```

Comment:
Another one is that all chains in a dependency set, and their lockbox, must have the same `adminOwner()`